internal/engine/wazevo/ssa/basic_block_sort.go

```diff
@@ -0,0 +1,23 @@
+//go:build go1.21
+
+package ssa
+
+import (
+	"slices"
+)
+
+func sortBlocks(blocks []*basicBlock) {
+	slices.SortFunc(blocks, func(i, j *basicBlock) int {
+		if j.ReturnBlock() {
+			return 1
+		}
+		if i.ReturnBlock() {
+			return -1
+		}
```

Comment:
Should the function return zero when `i.ReturnBlock() && j.ReturnBlock()` ?

I honestly don't know if it matters, but it could confuse the sorting algorithm if `f(i, h) == 1` but also `f(j, i) == 1`.

The comparison function has three states bascially for less than, equal, and greater than, which is slightly different from the "less" function that was historically used in the `sort` package.

Reply:
yeah it should return zero but it doesn't make any difference in the result at the end of the day. Would you mind raising a fix if you like? 🏃

Reply:
These things make me nervous, I've seen quick-sort turn into O(N^2) because the comparison function wasn't producing a coherent output.

Sending a PR 👍